src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Lambda/LambdaEventHelpers.cs

```diff
@@ -197,9 +197,9 @@ private static void TryParseSQSDistributedTraceHeaders(dynamic sqsEvent, ITransa
                     sqsHeaders.Add(messageAttribute.Key, messageAttribute.Value.Value);
                 }
             }
-            catch
+            catch (Exception e)
             {
-                // do nothing
+                agent.Logger.Log(Logging.Level.Debug, $"Caught exception in TryParseSQSDistributedTraceHeaders. Exception: {e.Message}, record.Body={(string)record.Body}");
```

Comment:
I don't think that we should record the message body, because it could result in customer data being stored in an unexpected location. I think that recording the exception is still valuable though.

Reply:
We need to be careful about flooding the logs with this message though. We should probably restrict the number of times it is recorded. For example, just log the first 3 errors (of any type).

Reply:
Good point. I hadn't thought about data leakage.

Reply:
It's hard to see how useful the log message could be without seeing the body of what's being deserialized.  I feel like we should have some kind of umbrella disclaimer that the agent may write logs that contain sensitive data and customers should be responsible for securing access to the log file on their system.  However, I'd rather get this PR put to bed.

Reply:
Well, there can be problems when customers have requirements that all data at rest must be encrypted. It's one thing to log metadata about what's happening. And it's another thing to log actual payloads.